plugins/woocommerce/includes/class-wc-coupon.php

```diff
@@ -1097,4 +1097,49 @@ public static function get_generic_coupon_error( $err_code ) {
 		// When using this static method, there is no $this to pass to filter.
 		return apply_filters( 'woocommerce_coupon_error', $err, $err_code, null );
 	}
+
+	/**
+	 * Get the coupon information that is needed to reapply the coupon to an existing order.
+	 * This information is intended to be stored as a meta value in the order line item corresponding to the coupon.
+	 *
+	 * The information returned is a JSON-encoded string of an array with the following coupon information:
+	 *
+	 * 0: Id
+	 * 1: Code
+	 * 2: Type, null is equivalent to 'fixed_cart'
+	 * 3: Amount
+	 * 4: The coupon grants free shipping? (present only if true)
+	 *
+	 * @return string A JSON string with information that allows the coupon to be reapplied to an existing order.
+	 */
+	public function get_short_info(): string {
+		$type = $this->get_discount_type();
+		$info = array(
```

Comment:
> Are you using array instead of hash for performance?

More than performance I did that to minimize the database space used.

> If so then note that this won't be extendible for 3pd to add their own data

And that's fine I think, this data is intended for internal usage and not for external code usage or extension (and that's why I added the extra field in the REST API endpoint, so that clients can simply forget about this). 3pd willing to add custom data should do so using a separate meta entry.

>  and discount amount is being stored as a separate meta key

What gets stored is the _actual amount_ discounted from the line, but not the _nominal amount_ of the coupon; both will be different if the coupon is of type "percent".

> Also note that we don't need to store the code or discount amount

Storing the code here is indeed redundant, but I think it's important for `coupon_info` to be "self-contained", especially as it's a replacement for `coupon_data` which contained the same information. It's a redundancy that we can afford in order to reduce confussion and increase robustness.

Reply:
That makes sense, can you please add this info as a comment, specifically that 3PD should add their data in a different field and that the store info is external.

> What gets stored is the actual amount discounted from the line, but not the nominal amount of the coupon; both will be different if the coupon is of type "percent".

I see, 👍 . Can you share an example of this difference, maybe we need to add some more documentation around it to clarify this difference?

Reply:
Ok, I have extended the comment of `get_short_info`. Anyway I was planning to publish an announcement post in the developer blog once this is released.